zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -982,6 +991,67 @@ where
         }
         .boxed()
     }
+
+    fn z_list_unified_receivers(
+        &self,
+        address: String,
+    ) -> BoxFuture<Result<unified_address::Response>> {
+        use zcash_address::unified::Container;
+
+        async move {
+            let (network, unified_address): (
+                zcash_address::Network,
+                zcash_address::unified::Address,
+            ) = zcash_address::unified::Encoding::decode(address.clone().as_str()).map_err(
+                |error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                },
+            )?;
+
+            let mut p2pkh = String::new();
+            let mut p2sh = String::new();
+            let mut orchard = String::new();
+            let mut sapling = String::new();
+
+            for item in unified_address.items() {
+                match item {
+                    zcash_address::unified::Receiver::Orchard(_data) => {
+                        let addr = zcash_address::unified::Address::try_from_items(vec![item])
+                            .expect("using data already decoded as valid");
+                        orchard = addr.encode(&network);
+                    }
+                    zcash_address::unified::Receiver::Sapling(data) => {
+                        let addr =
+                            zebra_chain::primitives::Address::try_from_sapling(network, data)
+                                .expect("using data already decoded as valid");
+                        sapling = addr.payment_address().unwrap_or_default();
+                    }
+                    zcash_address::unified::Receiver::P2pkh(data) => {
+                        let addr = zebra_chain::primitives::Address::try_from_transparent_p2pkh(
+                            network, data,
+                        )
+                        .expect("using data already decoded as valid");
+                        p2pkh = addr.payment_address().unwrap_or_default();
+                    }
+                    zcash_address::unified::Receiver::P2sh(data) => {
+                        let addr = zebra_chain::primitives::Address::try_from_transparent_p2sh(
+                            network, data,
+                        )
+                        .expect("using data already decoded as valid");
+                        p2sh = addr.payment_address().unwrap_or_default();
+                    }
```

Comment:
Optional: these all seem like implementations we might want to move to methods on zebra_chain::primitives::Address, but we can do that in a follow-up PR.